pkg/cache/resource_cache.go

```diff
@@ -229,11 +229,24 @@ func SaveResourceSHA(resource *unstructured.Unstructured, shaType SHAType) {
 	key := object.UnstructuredToObjMetadata(resource).String()
 	sha, _ := resourceCache.GetCacheEntry(key)
 	if err := sha.SetSHA(*resource, shaType); err == nil {
+		klog.ErrorS(err, "could not set SHA for resource", "key", key, "shaType", shaType)
 		sha.uid = string(resource.GetUID())
 		resourceCache.SetCacheEntry(key, sha)
 	}
 }
 
+func CommitManifestSHA(resource *unstructured.Unstructured) {
+	if !initialized {
+		klog.V(4).Info("resource cache not initialized")
+		return
+	}
+
+	key := object.UnstructuredToObjMetadata(resource).String()
+	sha, _ := resourceCache.GetCacheEntry(key)
+	sha.CommitManifestSHA()
+	resourceCache.SetCacheEntry(key, sha)
```

Comment:
do you actually need to set it back? I'm guessing it's just updating the sha in-place and it's already in the cache hash map right

Reply:
It might work without an explicit set, since the underlying fields are pointers. The `GetCacheEntry` returns an object so a shallow copy of the cache entry. Pointers should still reference the same memory though.

Reply:
It's probably fine to keep this as-is now to be safe